tests/utils/debug.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * The debug utility that ships with react native testing library does not work properly and
+ * has limited functionality. This is a better version of it that allows logging a subtree of
+ * the app.
+ */
+
+/* eslint-disable  no-console, testing-library/no-node-access, testing-library/no-debugging-utils, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-explicit-any */
+import type {NewPlugin} from 'pretty-format';
+import prettyFormat, {plugins} from 'pretty-format';
```

Comment:
@janicduplessis does `pretty-format` need to be added as a dev dependency?

Reply:
It is already included as part of testing-library/react-native, but I added it here since we use it directly now.

If that is a problem I can remove this debug util and add it again in a follow up. It has been very useful for debugging the ui tests.

Reply:
Another option would be to try to upstream this back to testing-library so the `screen.debug` utility works properly.

Reply:
I think this is probably ok for now, we've got bigger fish to fry imo